Auth.php

```diff
@@ -678,34 +674,43 @@ protected function addUser($email, $password, $params = [], &$use_email_activati
         }
 
         $password = $this->getHash($password);
+        if(strlen($this->config->table_params)==0){ 
+            //retain backwards compatibility
+            if (is_array($params)&& count($params) > 0) {
+                $customParamsQueryArray = [];
 
-        if (is_array($params)&& count($params) > 0) {
-            $customParamsQueryArray = [];
-
-            foreach($params as $paramKey => $paramValue) {
-                $customParamsQueryArray[] = ['value' => $paramKey . ' = ?'];
-            }
-
-            $setParams = ', ' . implode(', ', array_map(function ($entry) {
-                return $entry['value'];
-            }, $customParamsQueryArray));
-        } else { $setParams = ''; }
-
-        $query = "UPDATE {$this->config->table_users} SET email = ?, password = ?, isactive = ? {$setParams} WHERE id = ?";
-        $query_prepared = $this->dbh->prepare($query);
+                foreach($params as $paramKey => $paramValue) {
+                    $customParamsQueryArray[] = ['value' => $paramKey . ' = ?'];
+                }
 
-        $bindParams = array_values(array_merge([$email, $password, $isactive], $params, [$uid]));
+                $setParams = ', ' . implode(', ', array_map(function ($entry) {
+                    return $entry['value'];
+                }, $customParamsQueryArray));
+            } else { $setParams = ''; }
 
-        if (!$query_prepared->execute($bindParams)) {
-            $query = "DELETE FROM {$this->config->table_users} WHERE id = ?";
+            $query = "UPDATE {$this->config->table_users} SET email = ?, password = ?, isactive = ? {$setParams} WHERE id = ?";
+            $query_prepared = $this->dbh->prepare($query);
+            $bindParams = array_values(array_merge([$email, $password, $isactive], $params, [$uid]));
+            
+        } else {
+            try { 
+                $json_params=json_encode($this->appyFilterParams($params)); 
+            } catch (\Exception $e){
+                $this->dbh->rollBack(); 
+                $return['message']=$e->getMessage(); 
+                return $return; 
+            }
+            $query = "UPDATE {$this->config->table_users} SET email = ?, password = ?, isactive = ?, params = ?  WHERE id = ?";
             $query_prepared = $this->dbh->prepare($query);
+            $bindParams = array($email, $password, $isactive, $json_params, $uid);
+        }
 
-            $query_prepared->execute([$uid]);
+        if (!$query_prepared->execute($bindParams)) {
+            $this->dbh->rollBack(); 
             $return['message'] = $this->__lang("system_error") . " #04";
-
             return $return;
         }
-
+        $this->dbh->commit(); 
```

Comment:
How about checking the return value? In case of error the method returns false.

Reply:
what method? that part of the function is only when the data is inserted... therefore the commit...